Userland/Applications/SoundPlayer/SoundPlayerWidgetAdvancedView.cpp

```diff
@@ -142,6 +142,15 @@ void SoundPlayerWidgetAdvancedView::keydown_event(GUI::KeyEvent& event)
     if (event.key() == Key_Space)
         m_play_button->click();
 
+    if (event.key() == Key_S)
+        m_stop_button->click();
+
+    if (event.key() == Key_Up)
+        m_volume_slider->set_value(m_volume_slider->value() + m_volume_slider->page_step());
```

Comment:
I'm pretty confident there's a way of directly instructing a slider to "go up" or "go down".

Reply:
I looked for it but I haven't found it u-u 

I'm not sure if it exists because this is how the Slider mousedown event is implemented when you click inside of its rectangle but don't drag the knob, which is the exact same behavior I expected with the shortcuts. 

Maybe we could add it in a future PR?

https://github.com/SerenityOS/serenity/blob/43a800a83857709b0adca35f2f3a228680431ed0/Userland/Libraries/LibGUI/Slider.cpp#L104-L109

Reply:
I thought I had seen it in the Audio applet but there we're just forwarding mouse events on the applet to the slider. Maybe you can do something similar and create a fake scrolling event that you then forward to the slider?

Reply:
Okay, I will look into that (: 

I was implementing your idea and added some 'go_up()' and 'go_down()' methods to AbstractSlider. It seems that the 'set_value(value() + delta)' is a common pattern to move the slider and those methods should help make it simpler.

Do you think is a good solution?

https://github.com/elyse0/serenity/commit/73b3f622f664d875d435531554b0446fb66c407f

Reply:
Great idea, but generally if we make such a change, we also adopt all the existing code so that it uses the new abstracting functions, including the "real" event handler. If you do that, then I think it'll be easily accepted.

Reply:
Okaay, thanks n-n. I'll make that PR after this gets merged because these commits would appear again in the new one